std/bitmanip.d

```diff
@@ -886,8 +903,105 @@ public:
         return b;
     }
 
+    /**
+     * Sets all the bits in the `BitArray` to given value.
+     * Params:
+     * b = Value to which all the bits should be set.
+     */
+    void opIndexAssign(bool b) @nogc pure nothrow
+    {
+        if (b)
+        {
+            foreach (i; 0 .. dim)
+                _ptr[i] = ~size_t(0);
+            clearExtraBits();
+        }
+        else
+        {
+            _ptr[0 .. dim] = 0;
+        }
+    }
+
+    @system unittest
+    {
+        debug (bitarray) printf("BitArray.opIndexAssign.unittest\n");
+
+        auto ba = BitArray([0, 0, 1, 0, 1, 1, 0]);
+        auto b7true = BitArray([1, 1, 1, 1, 1, 1, 1]);
+        auto b7false = BitArray([0, 0, 0, 0, 0, 0, 0]);
+
+        ba[] = 0;
+        assert(ba == b7false);
+        ba[] = true;
+        assert(ba == b7true);
+
+        auto bb = BitArray([0]);
+        auto b1true = BitArray([1]);
+        auto b1false = BitArray([0]);
+        bb[] = 1;
+        assert(bb == b1true);
+        bb[] = 0;
+        assert(bb == b1false);
+
+        auto b70 = BitArray([1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+            1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]);
+        auto b70true = BitArray([1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+            1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+            1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+            1, 1]);
+        auto b70false = BitArray([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0]);
+        b70[] = true;
+        assert(b70 == b70true);
+        b70[] = false;
+        assert(b70 == b70false);
+        b70[] = true;
+        assert(b70 == b70true);
+    }
+
+    /**
+     * Toggles all the bits in the `BitArray` in place.
+     * Returns: Copy of this `BitArray` with all the bits flipped.
+     */
+    @property BitArray flip() @nogc pure nothrow
```

Comment:
Why return a new `BitArray` when all of the values are modified in place? Why would I want two `BitArray`s that point to the same data?